File: backend/app/models/party.py
from typing import Optional, List, Dict
from pydantic import BaseModel, Field


class Party(BaseModel):
    id: int
    name: str
    description: Optional[str] = None
    campaign_id: Optional[int] = Field(None, description="Associated campaign ID")
    characters: List[int] = Field(default_factory=list, description="IDs of characters in the party")
    level: int = Field(ge=1, le=20, default=1, description="Average party level")
    experience_points: int = Field(ge=0, default=0, description="Total experience points of the party")
    notes: Optional[str] = None
    image_url: Optional[str] = None
    icon_url: Optional[str] = None

File: frontend/lib/data/models/party.dart
class Party {
  final int id;
  final String name;
  final String? description;
  final int? campaignId;
  final List<int>? memberIds;
  final int level;
  final int experiencePoints;
  final String? notes;
  final String? imagePath;
  final String? iconPath;

  Party({
    required this.id,
    required this.name,
    this.description,
    this.campaignId,
    this.memberIds,
    this.level = 1,
    this.experiencePoints = 0,
    this.notes,
    this.imagePath,
    this.iconPath,
  });

  factory Party.fromJson(Map<String, dynamic> json) {
    return Party(
      id: json['id'] as int,
      name: json['name'] as String,
      description: json['description'] as String?,
      campaignId: json['campaign_id'] as int?,
      memberIds: json['member_ids'] != null ? List<int>.from(json['member_ids']) : null,
      level: json['level'] as int? ?? 1,
      experiencePoints: json['experience_points'] as int? ?? 0,
      notes: json['notes'] as String?,
      imagePath: json['image_path'] as String?,
      iconPath: json['icon_path'] as String?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'description': description,
      'campaign_id': campaignId,
      'member_ids': memberIds,
      'level': level,
      'experience_points': experiencePoints,
      'notes': notes,
      'image_path': imagePath,
      'icon_path': iconPath,
    };
  }
}

